src/documentation/io/authoring-overview.md

```diff
@@ -10,35 +10,80 @@ permalink: /documentation/io/authoring-overview/
 
 _A guide for users who need to connect to a data store that isn't supported by the [Built-in I/O Transforms]({{site.baseurl }}/documentation/io/built-in/)_
 
-> Note: This guide is still in progress. There is an open issue to finish the guide: [BEAM-1025](https://issues.apache.org/jira/browse/BEAM-1025).
 
 * TOC
 {:toc}
 
 ## Introduction
-TODO
+This guide covers how to implement I/O transforms in the Beam model. Beam pipelines use these read and write transforms to import data for processing, and write data to a store.
+
+Reading and writing data in Beam is a parallel task, and using `ParDo`s, `GroupByKey`s, etc... is usually sufficient. Rarely, you will need the more specialized `Source` and `Sink` classes for specific features. There are changes coming soon (`SplittableDoFn`, [BEAM-65](https://issues.apache.org/jira/browse/BEAM-65)) that will make `Source` unnecessary.
+
+As you work on your I/O Transform, be aware that the Beam community is excited to help those building new I/O Transforms and that there are many examples and helper classes.
 
-## Example I/O Transforms
-TODO
 
 ## Suggested steps for implementers
-TODO
+1. Check out this guide and come up with your design. If you'd like, you can email the [Beam dev mailing list]({{ site.baseurl }}/get-started/support) with any questions you might have. It's good to check there to see if anyone else is working on the same I/O Transform.
+2. If you are planning to contribute your I/O transform to the Beam community, you'll be going through the normal Beam contribution life cycle - see the [Apache Beam Contribution Guide]({{ site.baseurl }}/contribute/contribution-guide/) for more details.
+3. As you're working on your IO transform, see the [PTransform Style Guide]({{ site.baseurl }}/contribute/ptransform-style-guide/) for specific information about writing I/O Transforms.
+
 
 ## Read transforms
-TODO
+Read transforms take data from outside of the Beam pipeline and produce PCollections of data.
+
+For data stores or files types where the data can be read in parallel, you can think of the process as a mini-pipeline. This often consists of two steps:
```

Comment:
not sure if this is just referring to the extraneous 's' in "files types", or something else. I agree there's an extra s, "file types" (no s, with a space) seems the most correct.